clang/lib/CIR/Dialect/Transforms/TargetLowering/LowerFunction.cpp

```diff
@@ -386,14 +386,11 @@ LowerFunction::buildFunctionProlog(const LowerFunctionInfo &FI, FuncOp Fn,
       // the argument is used only to be stored in a alloca.
       Value arg = SrcFn.getArgument(ArgNo);
       assert(arg.hasOneUse());
-      for (auto *firstStore : arg.getUsers()) {
-        assert(isa<StoreOp>(firstStore));
```

Comment:
This assert is redundant, since `cast` should already be performing an assertion in asserts builds, so I took the opportunity to delete it.